app/reducers/traces.js

```diff
@@ -104,6 +108,43 @@ export default function (state = initialState, action) {
         watcher: action.watcher
       }
     }
+
+    case types.TRACE_UPLOAD_STARTED: {
+      const traces = _cloneDeep(state.traces)
+      const index = _findIndex(state.traces, t => t.id === action.id)
+      traces[index].uploading = true
+      return {
+        ...state,
+        traces
+      }
+    }
+
+    case types.TRACE_UPLOADED: {
+      const traces = _cloneDeep(state.traces)
+      const index = _findIndex(state.traces, t => t.id === action.oldId)
+      traces[index].pending = false
+      traces[index].uploading = false
+      traces[index].id = action.newId
+      if (!traces[index].properties) {
```

Comment:
My original idea was not correct. The error I was seeing was because the properties object doesn't exist sometimes when adding the id.

Reply:
Hmmm it seems a bit strange that the properties object would not exist - this maybe better to ensure a bit further up when the trace is being created. Let me look a bit closer. Thanks for the catch / fix.